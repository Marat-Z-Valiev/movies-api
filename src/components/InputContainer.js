import React from "react";
import Select from "react-select";
import Input from "./Input";
import Button from "./Button";
import InputContainerStyled from "./styles/InputContainerStyled";

const options = [
	{value: "movie", label: "Movie"},
	{value: "game", label: "Game"},
	{value: "series", label: "Series"}
];

const InputContainer = ({
	handleChange,
	handleClick,
	handleSelect,
	isLoading,
	isDisabled
}) => (
	<>
		<InputContainerStyled>
			<Input handleChange={handleChange} handleClick={handleClick} />
			<h2>Select category</h2>
			<Select
				className="select-category"
				defaultValue={{value: "movie", label: "Movie"}}
				options={options}
				onChange={handleSelect}
			/>
			<Button
				handleClick={handleClick}
				isLoading={isLoading}
				isDisabled={isDisabled}
			/>
		</InputContainerStyled>
	</>
);

export default InputContainer;
