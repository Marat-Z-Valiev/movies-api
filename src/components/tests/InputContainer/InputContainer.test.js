import React from "react";
import {shallow, configure} from "enzyme";
import toJSON from "enzyme-to-json";
import Adapter from "enzyme-adapter-react-16";
import InputContainer from "../../InputContainer";

configure({adapter: new Adapter()});

describe("InputContainer", () => {
	it("should render correctly", () => {
		const component = shallow(
			<InputContainer
				handleChange={() => {}}
				handleClick={() => {}}
				handleSelect={() => {}}
				isLoading={false}
				isDisabled={false}
			/>
		);
		expect(toJSON(component)).toMatchSnapshot();
	});
});
