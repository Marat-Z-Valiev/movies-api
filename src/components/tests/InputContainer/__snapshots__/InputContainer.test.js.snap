// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputContainer should render correctly 1`] = `
<Fragment>
  <styled.div>
    <Input
      handleChange={[Function]}
      handleClick={[Function]}
    />
    <h2>
      Select category
    </h2>
    <StateManager
      className="select-category"
      defaultInputValue=""
      defaultMenuIsOpen={false}
      defaultValue={
        Object {
          "label": "Movie",
          "value": "movie",
        }
      }
      onChange={[Function]}
      options={
        Array [
          Object {
            "label": "Movie",
            "value": "movie",
          },
          Object {
            "label": "Game",
            "value": "game",
          },
          Object {
            "label": "Series",
            "value": "series",
          },
        ]
      }
    />
    <Button
      handleClick={[Function]}
      isDisabled={false}
      isLoading={false}
    />
  </styled.div>
</Fragment>
`;
